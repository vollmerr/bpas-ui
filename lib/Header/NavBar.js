var _templateObject = _taggedTemplateLiteral(['\n  min-width: 100%;\n  background: ', ';\n  border-bottom: 5px solid ', ';\n'], ['\n  min-width: 100%;\n  background: ', ';\n  border-bottom: 5px solid ', ';\n']),
    _templateObject2 = _taggedTemplateLiteral(['\n  align-items: center;\n  display: ', ';\n  justify-content: space-between;\n  min-height: ', 'px;\n'], ['\n  align-items: center;\n  display: ', ';\n  justify-content: space-between;\n  min-height: ', 'px;\n']);

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';

import Container from '../Container';
import NavLogo from './NavLogo';
import NavLinks from './NavLinks';
import NavMobile from './NavMobile';

var Wrapper = styled.div(_templateObject, theme.color.grey.white, theme.color.accent.yellow);

var Content = styled.div(_templateObject2, function (props) {
  return props.isMobile ? 'block' : 'flex';
}, theme.size.navBar);

/** Renders the headers title bar */
function NavBar(_ref) {
  var links = _ref.links,
      _ref$isMobile = _ref.isMobile,
      isMobile = _ref$isMobile === undefined ? false : _ref$isMobile,
      _ref$mobileNavVisible = _ref.mobileNavVisible,
      mobileNavVisible = _ref$mobileNavVisible === undefined ? false : _ref$mobileNavVisible,
      onClick = _ref.onClick,
      onToggle = _ref.onToggle;

  var navLogoProps = {
    isMobile: isMobile,
    onClick: onClick
  };

  var navMobileProps = {
    onClick: onToggle
  };

  var navLinksProps = {
    links: links,
    isMobile: isMobile,
    onClick: onClick
  };

  var displayLinks = mobileNavVisible || !isMobile;

  return React.createElement(
    Wrapper,
    null,
    React.createElement(
      Container,
      null,
      React.createElement(
        Content,
        { isMobile: isMobile },
        React.createElement(NavLogo, navLogoProps),
        isMobile && React.createElement(NavMobile, navMobileProps),
        displayLinks && React.createElement(NavLinks, navLinksProps)
      )
    )
  );
}

NavBar.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    href: PropTypes.string,
    icon: PropTypes.string
  })),
  isMobile: PropTypes.bool,
  mobileNavVisible: PropTypes.bool,
  onClick: PropTypes.func
};

export default NavBar;