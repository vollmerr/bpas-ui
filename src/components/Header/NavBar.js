import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';

import Container from '../Container';
import NavLogo from './NavLogo';
import NavLinks from './NavLinks';
import NavMobile from './NavMobile';

const Wrapper = styled.div`
  min-width: 100%;
  background: ${theme.color.primaryInverse};
  border-bottom: 5px solid ${theme.color.second};
`;

const Content = styled.div`
  align-items: center;
  display: ${props => props.isMobile ? 'block' : 'flex'};
  justify-content: space-between;
  min-height: ${theme.size.navBar}px;
`;

/** Renders the headers title bar */
function NavBar({
  links,
  isMobile = false,
  mobileNavVisible = false,
  onClick,
  onToggle,
}) {
  const navLogoProps = {
    isMobile,
    onClick,
  };

  const navMobileProps = {
    onClick: onToggle,
  };

  const navLinksProps = {
    links,
    isMobile,
    onClick,
  };

  const displayLinks = mobileNavVisible || !isMobile;

  return (
    <Wrapper>
      <Container>
        <Content isMobile={isMobile}>
          <NavLogo {...navLogoProps} />
          {isMobile && <NavMobile {...navMobileProps} />}
          {displayLinks && <NavLinks {...navLinksProps} />}
        </Content>
      </Container>
    </Wrapper>
  );
}

NavBar.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  isMobile: PropTypes.bool,
  mobileNavVisible: PropTypes.bool,
  onClick: PropTypes.func,
};

export default NavBar;
