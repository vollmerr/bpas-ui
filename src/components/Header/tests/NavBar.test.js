import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../NavBar';

describe('<NavBar />', () => {
  it('renders correctly', () => {
    const tree = shallow(
      <NavBar />,
      { context: {history: {}} }
    );
    expect(tree).toMatchSnapshot();
  });

  xit('hides NavMobile when isMobile', () => {});

  xit('displays NavLinks when mobileNavVisible or NOT isMobile', () => {});

  xit('passes the correct props down', () => {});
});
