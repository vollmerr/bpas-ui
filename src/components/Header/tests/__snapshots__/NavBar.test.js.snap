// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<styled.div>
  <Container>
    <styled.div
      isMobile={false}
    >
      <NavLogo
        isMobile={false}
      />
      <NavLinks
        isMobile={false}
      />
    </styled.div>
  </Container>
</styled.div>
`;
